skrifa/src/outline/autohint/metrics.rs

```diff
@@ -68,6 +72,71 @@ pub(crate) struct UnscaledStyleMetrics {
     pub axes: [UnscaledAxisMetrics; 2],
 }
 
+/// The set of unscaled style metrics for a single font.
+///
+/// For a variable font, this is dependent on the location in variation space.
+#[derive(Debug)]
+pub(crate) enum UnscaledStyleMetricsSet {
+    Precomputed(Vec<UnscaledStyleMetrics>),
+    #[cfg(feature = "std")]
+    Lazy(RwLock<Vec<Option<UnscaledStyleMetrics>>>),
+}
+
+impl UnscaledStyleMetricsSet {
+    /// Creates a precomputed style metrics set containing all metrics
+    /// required by the glyph map.
+    pub fn precomputed(font: &FontRef, coords: &[F2Dot14], style_map: &GlyphStyleMap) -> Self {
+        let mut vec = Vec::with_capacity(style_map.metrics_count());
+        for style in style_map.metrics_styles() {
+            let metrics = super::latin::compute_unscaled_style_metrics(font, coords, style);
+            vec.push(metrics);
+        }
```

Comment:
any reason to prefer this to map/collect?

Reply:
`GlyphStyleMap::metrics_styles()` doesn't produce an exact size iterator and I'm trying to avoid over-allocating memory whenever possible in this code.

Reply:
That is non-obvious enough to merit a comment

Reply:
Will change to extend/map and add a comment noting why we don't collect